### ISSUE TITLE
Could you provide practical examples of Polygons with Holes?

### ISSUE THREAD

**Issue by renatodex** [goostengine/goost-examples#1]
Hello!

I'm using Goost for the first time, and I'm having some difficult drawing a Polygon with a hole in the middle
Do you think you could provide some practical examples?

I would like to do exactly this:
![image](https://user-images.githubusercontent.com/68507/90866248-00f02c80-e36a-11ea-829a-8d744454ec0c.png)

My issue is that even using the properties from the Goost merge method, my Polygon2D object still can't understand the polygon vertexes to draw the clipped polygon.

Here's a repository for reproduction:
https://github.com/renatodex/goost-clip-polygon-example

**Comment by Xrayez** [comment]
I have transferred this issue to a dedicated `goost-examples` repository now. The bug/enhancement is tracked in godotengine/godot#41447. There's no way to draw polygons with holes in Goost without re-implementing the `Polygon2D` node, so it's better that holes support is added directly in Godot's `Polygon2D` node instead. It may be feasible to implement some `Polygon2D` variant in Goost itself, but requires design work for something like this to be usable by many.

**Comment by renatodex** [comment]
Cool!
Thanks for the support on that @Xrayez !
As a temporary solution, I've ended up rendering multiple Polygon2D objects, so at least I can see them in the editor and use some of the features from Polygon2D.

I will definitively move into a more robust implementation as soon Godot team have time to explore this kind of enhancement,
And regardless of receiving Godot updates, which will probably take around 25 or 30 years (lol), the Decompose functionality from Goost module is already TRULLY AMAZING!

Regardless the Polygon2D interface, I think when writing something like:
```
...
var merge_result = GoostGeometry2D.decompose_multiple_polygons_into_convex([polygons])
var polygon2d = Polygon2D.new()
polygon2d.set_polygons(merge_result)
```

We should definitively expect Polygon2D to automatically Decompose nodes using the configured Engine, using the same pattern you use at the Goost module.
If Godot already have some kind of decomposing logic, I don't see why they could not use the same decomposition under the hood for `set_polygons` method.

This could of course be something you could support at Goost, but I would only focus on that now if, and only if, the Performance gain could be really really positive when drawing complex Polygons.

Thanks again!

**Comment by Xrayez** [comment]
Yes, I think I could implement this myself in Godot's `Polygon2D` directly in fact, but yeah the speed at which this can be done is bottleneck`ed by the review process, and it's likely that the implementation has to be agreed upon first (else my particular implementation may be rejected), so waiting for response in godotengine/godot#41447.

**Comment by Xrayez** [comment]
I'm currently implementing boolean operator nodes in 2D, `PolyNode2D` is converted to `Node2D`, so this should help debugging polygons with holes more or less easily (may as well solve your underlying use case, I presume), feedback welcome!
